internal/controller/nodedisruption_controller_test.go

```diff
@@ -171,6 +171,54 @@ var _ = Describe("NodeDisruption controller", func() {
 			})
 		})
 
-	})
+		Context("If a NodeDisruption's nodeSelector does not match any node", func() {
+			var (
+				nodeLabelsNoMatch = map[string]string{"testselect": "nope"}
+				createdDisruption = &nodedisruptionv1alpha1.NodeDisruption{}
+			)
 
+			BeforeEach(func() {
+				disruption := &nodedisruptionv1alpha1.NodeDisruption{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "nodedisruption.criteo.com/v1alpha1",
+						Kind:       "NodeDisruption",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      NDName,
+						Namespace: NDNamespace,
+					},
+					Spec: nodedisruptionv1alpha1.NodeDisruptionSpec{
+						NodeSelector: metav1.LabelSelector{MatchLabels: nodeLabelsNoMatch},
+					},
+				}
+				Expect(k8sClient.Create(ctx, disruption.DeepCopy())).Should(Succeed())
+			})
+			AfterEach(func() {
+				clearAllNodeDisruptionRessources()
+			})
+      // TODO: find how to change controller config
```

Comment:
I struggled for some time on this :(

Reply:
I found a solution in the retry branch

Reply:
Ok I started doing a similar thing based on https://github.com/jetstack/kubebuilder-sample-controller/blob/master/controllers/suite_test.go
Let's fix this once your PR is done